### DISCUSSION TITLE
how to know port number of local jupyter server with WSL?

### DISCUSSION THREAD

**Discussion by pnwork** [microsoft/vscode-jupyter#11859] [Questions and Answers] [answered]
I want to open jupyter tab in chrome to have second window for my vs code. I use WSL Ubuntu. Also sometimes I want to know how this server is doing in general (e.g. for debugging purposes).

**Comment by DonJayamanne** [comment, marked as answer]
>  Also sometimes I want to know how this server is doing in general (e.g. for debugging purposes).

We don't always start Juptyer in VS Code, this is only a rare condition (if some native modules fail to load).
Hence I'd say that ther'es no port that can necessarily be retrieved.

If however you'd like to access this purely for debugging purposes, then please check the output in the `Juptyer` output panel and you might (as mentioned we rarely do this) find logs indicating Jupyter was started and contains the Url.

**Comment by pnwork** [comment]
@DonJayamanne ok, if there is no actual jupyter app, only kernel - how can I output cell output to another window/console/terminal/etc if I am on WSL? use case: I have long jupyter notebook with prototype code; I debug bottom cell, but I am interested in the outputs of the bottom cell I started from. I redirected output now to vs code console, but I would like to have a second window for convenience (I have multiple monitors). for example. can i redirect to a console connected to the same kernel?

**Reply by DonJayamanne** [reply]
I'm sorry, but I don't quite understand the issue.
Are you trying to?
* Start a kernel in VS Code
* & then connect to that same kernel for debugging (running arbitrary code) in another place such as another notebook or interactive window

**Reply by pnwork** [reply]
Sort of yes. I wanted to have an output of errors in another window which I can move to another monitor OR/AND be able to have another view of jupyter notebook open in vs code (with local or remote server, on WSL).

I found this (raises exception on warning), works fine:
```
import warnings
warnings.filterwarnings('error')
```

**Reply by DonJayamanne** [reply]
> another view of jupyter notebook open in vs code (with local or remote server, on WSL).

When you say `another window`, do you mean something like this?
I've got the same notebook opened twice, side by side and code running against the same kernel
![Screen Shot 2022-11-02 at 10 06 57](https://user-images.githubusercontent.com/1948812/199359013-a9d28f02-224d-4a59-bde1-047077049df3.png)
